src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -177,12 +177,16 @@ public double getI2() {
 
     @Override
     public double getPermanentLimit1() {
-        return branch.getCurrentLimits1() != null ? branch.getCurrentLimits1().getPermanentLimit() * getBus1().getNominalV() / PerUnit.SB : Double.NaN;
+        return branch.getCurrentLimits1() != null && getBus1() != null
```

Comment:
I put that at branch violation detections level. See my previous commit. I did that in order to avoid calling permanent limit before. What is the best practice ?

Reply:
The soon-to-be reverted refactoring needed to also check it here.